helix-term/src/handlers/auto_save.rs

```diff
@@ -0,0 +1,87 @@
+use std::time::Duration;
+
+use anyhow::Ok;
+use arc_swap::access::Access;
+
+use helix_event::{register_hook, send_blocking};
+use helix_view::{
+    editor::DEFAULT_AUTO_SAVE_DELAY,
+    events::DocumentDidChange,
+    handlers::{lsp::AutoSaveEvent, Handlers},
+    Editor,
+};
+use tokio::time::Instant;
+
+use crate::{
+    commands, compositor,
+    job::{self, Jobs},
+};
+
+#[derive(Debug)]
+enum State {
```

Comment:
This type is also useless/unneded the AutoSaveHandler does not have any state. This was again copied from signature help. And you can just remove this and make `AutoSaveHandler` a unit struct/zst